src/proto/single/cmd.rs

```diff
@@ -244,22 +244,22 @@ impl FaktoryCommand for Push {
 
 // ----------------------------------------------
 
-pub struct PushBulk<'a>(&'a [Job]);
+pub struct PushBulk(Vec<Job>);
 
-impl<'a> Deref for PushBulk<'a> {
-    type Target = &'a [Job];
+impl Deref for PushBulk {
```

Comment:
We'll probably want to remove this and instead implement `Serialize` for this method, which we can do with something like:
```rust
impl<I> Serialize for PushBulk<I>
where
    I: ExactSizeIterator<Item = Job>,
{
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: Serializer,
    {
        let mut seq = serializer.serialize_seq(Some(self.0.len()))?;
        for e in &mut self.0 {
            seq.serialize_element(e)?;
        }
        seq.end()
    }
}
```

Reply:
I only managed to do with interior mutability. Please have a look once again.

Reply:
Ahhh, it's because `FaktoryCommand` takes `&self`... That's annoying. The `RefCell` approach won't work either if `PushBulk::issue` is ever called more than once. And `FaktoryCommand::issue` takes `&self` precisely to prevent excessive cloning.

Hmm hmm hmm. We _could_ change `FaktoryCommand::issue` to take `self` (no `&`), though that feels excessive. There may be a way to do it with an awful bound like `for<'a> &'a I: IntoIterator`, but I think probably just stick with the version you had where `PushBulk` holds a `Vec<Job>` for now. Sorry for the back and forth!

Reply:
No worries, reverted.